pkg/controller/migmigration/verify.go

```diff
@@ -0,0 +1,329 @@
+				t.addVerificationDetails(podManager.GetKind(), ownerNamespacedName, failingPods)
+				return VerificationFailed, err
+			}
+			return VerificationProgressing, nil
+		}
+	}
+	return VerificationSucceeded, nil
+}
+
+func (t *Task) verifyDaemonSets(client k8sclient.Client, options *k8sclient.ListOptions) (int, error) {
+	daemonSetList := v1beta1.DaemonSetList{}
+
+	err := client.List(context.TODO(), options, &daemonSetList)
+	if err != nil {
+		return VerificationFailed, err
+	}
+
+	for _, ds := range daemonSetList.Items {
+		// All expected nodes are running the pods
+		if ds.Status.DesiredNumberScheduled == ds.Status.NumberReady {
+			continue
+		}
+
+		// Check the state of pods
+		failingPods, err := t.checkPodsHealth(client, options, ds.GetName())
+		if err != nil || len(failingPods) > 0 {
+			ownerNamespacedName := types.NamespacedName{
+				Namespace: ds.GetNamespace(),
+				Name:      ds.GetName(),
+			}
+			t.addVerificationDetails(ds.TypeMeta.Kind, ownerNamespacedName, failingPods)
+			return VerificationFailed, err
+		}
+		return VerificationProgressing, nil
+	}
+
+	return VerificationSucceeded, nil
+}
+
+func (t *Task) verifyJobs(client k8sclient.Client, options *k8sclient.ListOptions) (bool, error) {
+	jobList := batchv1.JobList{}
+
+	err := client.List(context.TODO(), options, &jobList)
+	if err != nil {
+		return false, err
+	}
+
+	for _, job := range jobList.Items {
+		if job.Status.Failed == 0 {
+			continue
+		}
+
+		// Check the state of pods
+		failingPods, err := t.checkPodsHealth(client, options, job.GetName())
+		if err != nil || len(failingPods) > 0 {
+			ownerNamespacedName := types.NamespacedName{
+				Namespace: job.GetNamespace(),
+				Name:      job.GetName(),
+			}
+			t.addVerificationDetails(job.TypeMeta.Kind, ownerNamespacedName, failingPods)
+			return false, err
+		}
+		break
+	}
+	return true, nil
+}
+
+// checkPodsHealth would collect, display and notify about the pod health in a specific namespace
+func (t *Task) checkPodsHealth(client k8sclient.Client, options *k8sclient.ListOptions, ownerName string) ([]string, error) {
+	failingPods := []string{}
+	resourceNames := []string{ownerName}
+	podList := corev1.PodList{}
+	eventList := corev1.EventList{}
+
+	err := client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return failingPods, err
+	}
+
+	err = client.List(context.TODO(), options, &eventList)
+	if err != nil {
+		return failingPods, err
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodFailed {
+			failingPods = append(failingPods, fmt.Sprintf("Pod:%s/%s", pod.Namespace, pod.Name))
+			resourceNames = append(resourceNames, pod.Name)
+			continue
+		}
+
+		// Consider suspicious containers as an error state
+		podContainersStatuses := append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...)
+		for _, containerStatus := range podContainersStatuses {
+			if !containerStatus.Ready && containerStatus.RestartCount >= CrashLoopBackOffMargin {
```

Comment:
I'm okay with this. We are saying ONLY if the pod is not ready will we do a check to see if there are more than 3 restarts.